tests/NgTests/StorageFactoryTests.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.ComponentModel;
+using Ng;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Xunit;
+
+namespace NgTests
+{
+    public class StorageFactoryTests
+    {
+        [Theory]
+        [Description("The azure commpressed factory should compress the content.")]
+        [InlineData("http://localhost/reg", "testAccount", "ZiWrbTMNTApZxLeWB7a0yN52gJj+ZlGE0ipRi9PaTcn9AU4epwvsngE5rLSMk9TwpazxUtzeyBnFeWFAhfkGpw==", "testContainer", "testStoragePath", "azure")]
+        public void AzureCommpressedFactory(string storageBaseAddress,
+                                            string storageAccountName,
+                                            string storageKeyValue,
+                                            string storageContainer,
+                                            string storagePath,
+                                            string storageType)
+        {
+            Dictionary<string, string> arguments = new Dictionary<string, string>()
+            {
+                { Arguments.CompressedStorageBaseAddress, storageBaseAddress },
+                { Arguments.CompressedStorageAccountName, storageAccountName },
+                { Arguments.CompressedStorageKeyValue, storageKeyValue },
+                { Arguments.CompressedStorageContainer, storageContainer },
+                { Arguments.CompressedStoragePath, storagePath },
+                { Arguments.StorageType, storageType},
+                { Arguments.UseCompressedStorage, "true"}
+            };
+
+            StorageFactory factory =  CommandHelpers.CreateCompressedStorageFactory(arguments, true);
+            AzureStorageFactory azureFactory = factory as AzureStorageFactory;
+            // Assert
+            Assert.True(azureFactory!=null);
```

Comment:
For this and subsequent tests you can use `Assert.Null(...)` or `Assert.NotNull(...)` instead of `Assert.True(...)` and `Assert.False(...)`.